import.go

```diff
@@ -78,10 +81,21 @@ func (i *Importer) writeNode(node *Node) error {
 
 	i.batchSize++
 	if i.batchSize >= maxBatchSize {
-		if err := i.batch.Write(); err != nil {
+		// Wait for previous batch.
+		var err error
+		if i.inflightCommit != nil {
+			err = <-i.inflightCommit
+			i.inflightCommit = nil
+		}
+		if err != nil {
 			return err
 		}
-		i.batch.Close()
+		result := make(chan error)
```

Comment:
A comment on why this is unbuffered would be helpful to clarify intent here. I would have expected it to be size 1 on my first read here, but there is probably a subtle reason why unbuffered is more appropriate.

Reply:
I didn't intent anything subtle here: I use a buffered channel when I need one, otherwise not. What's the use of a buffered channel here?